quint/src/effects/base.ts

```diff
@@ -43,6 +43,19 @@ export type Effect =
   | ArrowEffect
   | QuantifiedEffect
 
+/*
+ * An effect scheme, listing universally quantified variables (for names
+ * refering to Variables) and effect variables (for names refering to Effects)
+ */
+export type EffectScheme = {
+  /* The effect */
+  effect: Effect,
+  /* Universally quantified names refering to Effects */
+  effectVariables: Set<string>,
+  /* Universally quantified names refering to Variables */
+  variables: Set<string>,
```

Comment:
these are state variables, right? Maybe let's call them `stateVariables` then

Reply:
Kind of. They can be a generalization of state variables as well, and I'm already using the `stateVariables` name to refer to actual concrete state variables, so that would be equally confusing, I think. I'll try to improve these names in a follow-up PR if that's OK: https://github.com/informalsystems/quint/pull/658#discussion_r1119053458